### PR TITLE
Detect if Interface is running on an android testing device

### PR DESCRIPTION
**Pull request by gcalero**
This PR detect if Android Interface is running in testing mode based on the presence of a file in /sdcard/_hifi_test_device.txt. This replace the environment variables that we cannot set in android.

**Test Plan**
1. When the file is present in the device data sent to clearhistory includes the field tester=true
2. When the file is not in the device that field is sent as false
3. The file must be persistent (not created but not deleted by the app) even after uninstall + reinstall the app

**UPDATE**: you’ll need to check either clearstory or metabase to query the data
**UPDATE**: Manuscript case 19747

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8302/app_12718_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8302/HighFidelity-Beta-PR12718-288f69ad6ad5ee63fc97e1a70e6bb9b5805c6695.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8302/HighFidelity-Beta-PR12718-288f69ad6ad5ee63fc97e1a70e6bb9b5805c6695.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1281,11 +1284,20 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
...
     static const QString TESTER = "HIFI_TESTER";
+
+    bool isTester = false;
+#if defined (Q_OS_ANDROID)
+    // Since we cannot set environment variables in Android we use a file presence
+    // to denote that this is a testing device
+    QFileInfo check_tester_file(TESTER_FILE);
+    isTester = check_tester_file.exists() && check_tester_file.isFile();
+#endif
+    QProcessEnvironment::systemEnvironment().insert(TESTER, "This Value is Great");
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: Hahaha this looks like it's left over from testing.

Comment (gcalero) [reply interface/src/Application.cpp]: Ops... removing...

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8653/app_12718_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8653/HighFidelity-Beta-PR12718-6ccffb3c4dac878cda1b2f2ccad499f3c98c1b1f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8653/HighFidelity-Beta-PR12718-6ccffb3c4dac878cda1b2f2ccad499f3c98c1b1f.exe

**Comment by conklin94122** [comment]
@SamGondelman @gcalero We need a test plan and this PR has conflicts that need to be resolved.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14736/HighFidelity-Beta-PR12718-344774d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14736/HighFidelity-Beta-PR12718-344774d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14736/HighFidelity-Beta-Interface-PR12718-344774d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14736/HighFidelity-Beta-PR12718-344774d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14736/HighFidelity-Beta-Interface-PR12718-344774d.exe

**Comment by shanzzam** [comment]
This should be the PR for this one, correct? 
https://highfidelity.manuscript.com/f/cases/edit/19747/Exclude-testers-from-our-data-on-Android-devices

**Comment by shanzzam** [comment]
@gcalero If this PR fits the above ticket, can you update with a test plan, please.

**Comment by qligmaloney** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15041/HighFidelity-Beta-PR12718-344774d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15041/HighFidelity-Beta-Interface-PR12718-344774d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15041/HighFidelity-Beta-PR12718-344774d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15041/HighFidelity-Beta-Interface-PR12718-344774d.exe

**Comment by qlimnavarro** [comment]
Build verified runs on all platforms. No further pre-merge testing required.

**Pull request merged**